pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+)
+
+type PodWebhook struct{}
+
+// SetupWebhook configures the webhook for pods.
+func SetupWebhook(mgr ctrl.Manager, _ ...jobframework.Option) error {
+	wh := &PodWebhook{}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&corev1.Pod{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+func fromObject(o runtime.Object) *Pod {
+	return (*Pod)(o.(*corev1.Pod))
+}
+
+// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=ignore,sideEffects=None,groups="",resources=pods,verbs=create,versions=v1,name=mpod.kb.io,admissionReviewVersions=v1
+
+//TODO: check if we can use namespace/object selectors to skip this webhook
```

Comment:
I think we should. We want to be able to support manageJobsWithoutQueueName, but system pods should be excluded.
But let me finish the design before taking action.